core/arch/arm/plat-rcar/main.c

```diff
@@ -82,9 +89,23 @@ unsigned long plat_get_aslr_seed(void)
 
 	return seed;
 }
+#endif
 
 void main_init_gic(void)
 {
+#ifdef CFG_ARM_GICV3
+	vaddr_t gicd_base;
+
+	gicd_base = core_mmu_get_va(GICD_BASE, MEM_AREA_IO_SEC,
+				    GIC_DIST_REG_SIZE);
+
+	if (!gicd_base)
+		panic();
+
+	/* Initialize GIC */
+	gic_init(&gic_data, 0, gicd_base);
+	itr_init(&gic_data.chip);
+#else
 	vaddr_t gicc_base;
```

Comment:
This can be simplified to have less duplicated code with something like:
``` C
vaddr_t gicc_base = 0;
vaddr_t gicd_base = 0;

gicd_base = ...;
if (!gicd_base)
          panic();
if (!IS_ENABLED(CFG_ARM_GICV3)) {
          gicc_base = ...;
          if (!gicc_base)
                    panic();
}
...

Reply:
Thanks for the suggestion. Reworked and moved all the GIC-related changes to the first patch.

Looks like all boards implement GIC initialization in the same way. Maybe it is worth to provide default initialization function? I can prepare patch.

Reply:
Sure, let's see what you have in mind.